packages/jaeger-ui/src/utils/filter-spans.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import type { KeyValuePair, Span } from '../types/trace';
+
+export default function filterSpans(textFilter: string, spans: ?(Span[])) {
+  if (!spans) return null;
+
+  // if a span field includes at least one filter in includeFilters, the span is a match
+  const includeFilters = [];
+
+  // values with keys that include text in any one of the excludeKeys will be ignored
+  const excludeKeys = [];
+
+  // split textFilter by whitespace, remove empty strings, and extract includeFilters and excludeKeys
+  textFilter
+    .split(/\s+/)
+    .filter(Boolean)
```

Comment:
This is pre-existing code that I moved from a component into this util for re-usability. I tweaked the `split` and removed a `trim` in response to @tiffon . Subsequently, I do not believe this `filter` is necessary as `split` will not return any falsy (empty) strings.

Reply:
`textFilter.trim().split(/\s+/).forEach(/* ... */);` would be slightly better. I will change and make sure white space management is tested.